torch/distributed/device_mesh.py

```diff
@@ -652,6 +690,17 @@ def get_coordinate(self) -> Optional[List[int]]:
             """
             return self._coordinate_on_dim if self._coordinate_on_dim else None
 
+        def _flatten(self) -> "DeviceMesh":
+            """
+            Returns a 1D DeviceMesh by flattening the current DeviceMesh.
+            """
+            if not self.mesh_dim_names:
+                raise RuntimeError(
+                    "Cannot flatten a DeviceMesh without mesh_dim_names!"
+                )
+
+            return _mesh_resources.create_flatten_mesh(self)
```

Comment:
Are we going to cache the flattened mesh or this will be done in the following PR?

Reply:
Will include it in the next PR.